clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -303,6 +302,19 @@ private void throwIfOffsetOutOfRange() throws OffsetOutOfRangeException {
     }
 
     /**
+     * If any topic from previous fetchResponse contatains Authorization error, throw ApiException.
+     * @throws ApiException
+     */
+    private void throwIfUnauthorized() throws ApiException {
```

Comment:
Just to check my understanding, we wouldn't actually expect to hit this case unless the user has Describe access, but not Read access on any of its subscribed topics. If the user doesn't have Describe access, then we wouldn't be able to fetch metadata for the topic and we therefore wouldn't do any fetches. In that case, the user generally wouldn't see any errors because NetworkClient just logs any errors in metadata fetches. Is that expected behavior?